### DISCUSSION TITLE
NPWD 2.0 community ideas

### DISCUSSION THREAD

**Discussion by itschip** [project-error/npwd#930] [Ideas]
Hi! Now that we are slowly starting the development of the next major version of NPWD, 2.0, we would like to hear your input on what you've missed in v1 or something that could be done better. This includes apps, API design or other aspects of the phone/codebase.

Please do not ask for help here 👍🏻 

We're happy to hear from you!

**Comment by jeremy489** [comment]
I know alot of us use NPWD because its not reliant on a set framework, so my suggestion is to keep it that way! maybe making it where video calls could be a thing, possibly a few more social medias like snapchat or something.

**Comment by mikigoalie** [comment]
**Ideas**
Customs ringtones, music player with surrounding able to hear it _(youll probs deny it)_
Integrated option for sim cards _(prob would require ox inv stash function)_
Although I don't like it, alot of people ask for IG on my server
Option to block caller
Dark chat / reddit / group chats
Group calls? Can imagine it would be cool & unique feature
**Things to improve**
Enhance icons, rework twitter functions (profile, retweeting)

The latest versions of npwd got incredible improvements and generally, npwd is best phone atm, can't wait what you have prepared in v2 <3

**Reply by BombayV** [reply]
_Integrated option for sim cards_
**This is a heavy framework integration feature, therefore probably won't add it.**

_Although I don't like it, alot of people ask for IG on my server_
**As mentioned in the documentation. "Too similar already to Twitter and only used for images. Not worth the dev time."**

**Comment by ENC4YP7ED** [comment]
1. Garage-App (Luke_Garage). Where you can Locate your Vehicles and pay for the Parkinglot Payment.
2. More UI Performance
3. Let the Phone have 2 versions : Samsung Galaxy s22 Ultra version (Android) and a Iphone 14 plus max (ios16)
4. Add Notes-App
5. Calculator-App
6. Make Use Of ios16 Dynamic Island as Phonenotify
7. Framework Integrations (esx,qb.qbox,ox,nd) like a Bill-App for Paying Bills and a Jobcenter / Work-App for Applying for a Job or If you are in a Job than there would be an Jobannouncement channel and a Aktive Workerlist where you also can Fire people and uprank them.
8. (General Idea) Create A Carradio with Soundmuffiling when all doors are closed and Located Sound so that you can really hear where the music is coming from.
9. Airdrop
10. For Nr. 8 Carmusic-App so that you can configure Songs on Youtube/Soundcloud/Spotify.App and play it in the Car with all the other Features from Nr.8

**Reply by RuthlessScout** [reply]
Few of those functions can be added. There is a github with community made apps for NPWD

**Reply by BombayV** [reply]
Some of these suggestions are really iOS based (dynamic island, ios version, airdrop - 3, 6, 9). These will most likely never be implemented as NPWD is focused on being unique unlike most paid phones in the market.

2) The only app with an existing problem is the matchmaking app. Apart from that, everything is pretty much optimized. You can also try using this. 
![Image](https://cdn.discordapp.com/attachments/760985469151346720/1043701845383721101/image.png)
4) Already exists.
5) Already exists.
7) NPWD is a standalone phone. The phone at its current state provides a descriptive documentation on how to integrate your framework with our phone.
8/10) - As mentioned in the documentation, any kinds of Music Apps will not be integrated as this can lead to DMCA or copyright issues.

**Comment by Kathou99** [comment]
I know is a lot but to have a video call with RTC technology,this will be awsome.

The next idea is to load the app inside the npwd folder for better control of the app because when  you have 20 app is being difficult to keep a good optimisation 😆


And my last option is to offer a custom way to build our own api to store picture,in my case it's was the most stable options

Thank again for this awsome phone 📱

By the way I love android and Samsung,so the fact is no an apple is so great :)

**Comment by BlacknGoldy** [comment]
Information APP:
Name
ID
Money
Bank
....
---------------
**and PING APP:
can share the location with ID nr.**

**Reply by LukeWasTakenn** [reply]
> PING APP

Isn't this already built into the messages app?

**Comment by ghost** [comment]
Dispatch App or PD and Stuff as Sticky Contact to Everyones Phone

**Reply by BombayV** [reply]
Your suggestion is very framework based as it requires jobs. NPWD is a standalone phone therefore any job-related app will never occur.

**Reply by CeebDev** [reply]
You can just make "hasAccess" in app config, and all the framework stuff about job is outside NPWD, our script can just set hasAccess on this app based on our each framework
like : `exports.npwd:giveAccess("bank", false)`
or `exports.npwd:giveAccess("dispatch", true)`

**Comment by Cali-Suprabanana** [comment]
Hi,
-Having notifications on app icon when u're connecting (calls / contacts)
-Having possibility to put notification on app icon
- Better way to refresh apps, right now external apps can be refresh with going back to main menu and going back on app phone, so refreshing app in live can be good
- Custom prop for phone ?
- Easier way to manage phone ring via config file
- Flashlight on phone ?
- Airplane mode to not receive phone call
- Anonymous phonecall
- Easily change from os clock to game clock
- Stopwatch / Alarm 

It's more about functionnality than new apps.
Tyvm for u're awesome work.

**Reply by Ekinoxx0** [reply]
- Custom prop for phone : already possible ? Easy to change
 - Easier way to manage phone ring via config file : seems like this suggestion was refused because of some feedback from the Cfx team, i would agree there is more to it rather than just using the game system, having the two system available for servers or players to choose from
 
 I agree with the rest of your suggestions

**Reply by Cali-Suprabanana** [reply]
Yeah, not that hard to change the custom prop for phone, but could be gr8 to have one working with actual artistic direction.

Didn't knew for phone ring that was refused because of somes feedback, good to know, anyway, that was something for peoples that don't really know how to change these kind of things, and it could've been somes details that peoples could've liked. If it's refused, if peoples want to change it themself, they'll have to work on it.

**Comment by RAXIC7** [comment]
Npwd should be an all-in-one phone. Phone separate script, banking app separate script, garage app separate script shouldn't be. Except these the document should be kept up-to-date. It should be a user-friendly and simple script. For documents such as photos and audio it should be able to be saved in places other than your api. For example, I want to save the photos and sounds to the local, such an option should be provided. Should work fully with frameworks. Separate scripts should not be installed for frameworks.

**Reply by BombayV** [reply]
1) _It should be a user-friendly and simple script._ 
**NPWD is fully structured phone written in TypeScript. The phone itself is very user-friendly and developer friendly to those who have at least some knowledge in reading documentation and understand TypeScript. However, this is not the case for most of the FiveM community. A phone such as this cannot be a _simple_ script as it has been built over the years.**

2) _Npwd should be an all-in-one phone. Phone separate script, banking app separate script, garage app separate script shouldn't be._
**As mentioned before, NPWD is a project that keeps expanding over time. Having to maintain everything inside the phone causes issues, especially when integrating such large projects like the banking or garage app. The use of externals apps help us keep track of issues/bugs along with facilitating the development experience.**

3) _For documents such as photos and audio it should be able to be saved in places other than your api._
**You are free to use your own API such as Imgur or whatever other option you want. We encourage the use of our own services as they provide an easier way for us to help you and resolve issues quicker. **

4) _Should work fully with frameworks._
**NPWD is and will always be standalone. There is an extensive documentation on how to integrate with the most popular frameworks, even if you don't use one of these.**

5) _Separate scripts should not be installed for frameworks._
**I don't know which separate scripts you're talking about unless you mean the external apps. This was done for a better developer experience which facilitates the development of new apps without having to work on the phone itself.**

**Comment by Ekinoxx0** [comment]
Here are a few of my notes :
 - Dont fail on things : In a lot of apps, the user is prompt to errors that just shouldn't exist in a User Experience way and errors could resolve themself, an exemple : If a conversation already exist, we shouldn't refuse the creation, but redirect to the correct message conversation
- Fluid navigation flow, currently the phone is created around the "button way", click a button, expect something to happen, most modern OS (android / ios) now propose movement navigation with sliding on borders, return to home by sliding up, go backward with sliding right
  
 - Twitter as standalone/external app, even if we could considerate it as part of the phone, it's something that doesn't touch any main OS function and would lead the way for people to make their own social media apps
 - Matchmaker as a standalone/external app, same arguments as Twitter
 - Notes as standalone ? This also mean we need to allow 3rd app to integrate in OS apps like Messages by providing a template for "Sharing" notes, sharing Twitter profiles link, etc..
 
 - Major remake of Matchmaker, for now the apps seems very usable, but not very friendly for most users (from roleplayers feedback)
Some of the feedback from players : "Profile page is ugly, very basic with just a list field to fill", "Conversation are not easy to find", "Conversation should be more like the Messages app", this is very opinionated, but it seems to me this app was made because of community demand and the Projet Error team made it just to fill the demand (I would totally understand if no one in the team liked the app and didn't want to work on it, it happens)

 - Airdrop : Sharing number, files, photos with people around you in the world
 - Flashlight : Button to display a light in the world
 - Full screen zoom in photo, for now the zoom is a nice touch, but people seems to want to see the full details of every photo on any app (This includes Messages, Twitter, Gallery)
- Airplane mode (pretty much explains itself)
- 3rd party settings in settings app ? (Allow 3rd party apps to add buttons, checkbox, like twitter does)

Anyway, much love to the team for all the work, to be honest I didn't believe you were still motivated to make a V2, because most of the people you will see on discord will be people complaining on stuff, and you will almost never see people that just use this phone daily without any problem

**Comment by mikigoalie** [comment]
Another suggestions
Shared twitter accounts. Add multiple users to manage once acc

**Comment by themarwik** [comment]
maby a discord app? with channels, and supports that you can buy a channel name, news paper support for adding a news app

**Comment by uShifty** [comment]
Maybe for 2.0 have a svelte+vite app template

**Comment by gamenew09** [comment]
Notifications API should only have one function for creating notifications. Having to create a System notification using `createSystemNotification` for an app notification just to have an accept and reject button seems kinda scuffy.

**Comment by RAXIC7** [comment]
You should take apps developed by independent developers and make npwd's own default apps. Example:

npwd_crypto
npwd_adwertisements
npwd_race etc.

and you should also provide these apps with support for esx and qb core frameworks.

**Comment by DubStepMad** [comment]
Here are a few suggestions that I feel would be nice to implement into v2.0.

Since the intergration between developers can be hard, I fully understand some of these might not be possible or have limitations.

**Intergrations:**
- Codesign's Garage system: [Found Here](https://codesign.pro/package/4206352)
This would be nice to have an app on the phone to view personal cars and shared ones between groups (mainly pd and ems).
The app would also allow the player to locate the vehicle, tranfer ownership and possibly more things.

- Codesign's Multicharacter: [Found Here](https://codesign.pro/package/4206340)
Full support of their multicharacter script, this isn't too hard their script actually changes any columns named `identifier` to have the correct length but they do prefix with `Char1:`, `Char:2`, etc.

- OKOK's Banking: [Found Here](https://okok.tebex.io/package/5126428)
This would allow the user to manage their personal or group accounts.
Features like sending money, viewing transactions and more.

**Comment by RockySouthpaw** [comment]


**Reply by TonybynMp4** [reply]
welp it seems that having a team of very busy devs and no other contributors really killed it sadly 😢
